From 52af685cdad5af1e586fae9b19dc514882d94e26 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 25 Apr 2019 23:37:59 +0200
Subject: [PATCH 1/1] efi_loader: use upper case for GUIDs

This patch adjusts the GUID output in the device path to text protocol
to upper case to match the UEFI SCT implementation.

This changes the vendor node and the GPT GUID in the media node in device
paths.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_device_path_to_text.c   | 4 ++--
 lib/efi_selftest/efi_selftest_devicepath.c | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/lib/efi_loader/efi_device_path_to_text.c b/lib/efi_loader/efi_device_path_to_text.c
index 96fd08971b..3d4b9659db 100644
--- a/lib/efi_loader/efi_device_path_to_text.c
+++ b/lib/efi_loader/efi_device_path_to_text.c
@@ -62,7 +62,7 @@ static char *dp_hardware(char *s, struct efi_device_path *dp)
 	case DEVICE_PATH_SUB_TYPE_VENDOR: {
 		struct efi_device_path_vendor *vdp =
 			(struct efi_device_path_vendor *)dp;
-		s += sprintf(s, "VenHw(%pUl)", &vdp->guid);
+		s += sprintf(s, "VenHw(%pUL)", &vdp->guid);
 		break;
 	}
 	default:
@@ -190,7 +190,7 @@ static char *dp_media(char *s, struct efi_device_path *dp)
 			}
 		case SIG_TYPE_GUID:
 			s += sprintf(
-				s, "HD(%d,GPT,%pUl,0x%llx,0x%llx)",
+				s, "HD(%d,GPT,%pUL,0x%llx,0x%llx)",
 				hddp->partition_number, sig, start, end);
 			break;
 		default:
diff --git a/lib/efi_selftest/efi_selftest_devicepath.c b/lib/efi_selftest/efi_selftest_devicepath.c
index 4ce3fad895..8ba2a6c8e4 100644
--- a/lib/efi_selftest/efi_selftest_devicepath.c
+++ b/lib/efi_selftest/efi_selftest_devicepath.c
@@ -328,7 +328,7 @@ static int execute(void)
 	}
 	if (efi_st_strcmp_16_8(
 		string,
-		"/VenHw(dbca4c98-6cb0-694d-0872-819c650cbbb1)/VenHw(dbca4c98-6cb0-694d-0872-819c650cbba2)")
+		"/VenHw(DBCA4C98-6CB0-694D-0872-819C650CBBB1)/VenHw(DBCA4C98-6CB0-694D-0872-819C650CBBA2)")
 	    ) {
 		efi_st_printf("dp2: %ps\n", string);
 		efi_st_error("Incorrect text from ConvertDevicePathToText\n");
@@ -378,7 +378,7 @@ static int execute(void)
 		return EFI_ST_FAILURE;
 	}
 	if (efi_st_strcmp_16_8(string,
-			       "/VenHw(dbca4c98-6cb0-694d-0872-819c650cbbc3)")
+			       "/VenHw(DBCA4C98-6CB0-694D-0872-819C650CBBC3)")
 	    ) {
 		efi_st_printf("remaining device path: %ps\n", string);
 		efi_st_error("LocateDevicePath: wrong remaining device path\n");
-- 
2.20.1

